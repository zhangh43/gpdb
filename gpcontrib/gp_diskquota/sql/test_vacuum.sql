-- Test vacuum full
CREATE SCHEMA s6;
SELECT diskquota.set_schema_quota('s6', '1 MB');
SET search_path TO s6;
CREATE TABLE a (i int);
CREATE TABLE b (i int);
INSERT INTO a SELECT generate_series(1,100000000);
SELECT pg_sleep(5);
-- expect insert fail
INSERT INTO a SELECT generate_series(1,10);
-- expect insert fail
INSERT INTO b SELECT generate_series(1,10);
DELETE FROM a WHERE i > 10;
VACUUM FULL a;
SELECT pg_sleep(5);
-- expect insert succeed
INSERT INTO a SELECT generate_series(1,10);
INSERT INTO b SELECT generate_series(1,10);

DROP TABLE a, b;
RESET search_path;
DROP SCHEMA s6;

