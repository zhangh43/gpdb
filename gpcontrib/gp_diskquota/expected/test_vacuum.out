-- Test vacuum full
CREATE SCHEMA s6;
SELECT diskquota.set_schema_quota('s6', '1 MB');
 set_schema_quota 
------------------
 
(1 row)

SET search_path TO s6;
CREATE TABLE a (i int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
CREATE TABLE b (i int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
INSERT INTO a SELECT generate_series(1,100000000);
ERROR:  schema's disk space quota exceeded with name:s6
SELECT pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

-- expect insert fail
INSERT INTO a SELECT generate_series(1,10);
ERROR:  schema's disk space quota exceeded with name:s6
-- expect insert fail
INSERT INTO b SELECT generate_series(1,10);
ERROR:  schema's disk space quota exceeded with name:s6
DELETE FROM a WHERE i > 10;
VACUUM FULL a;
SELECT pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

-- expect insert succeed
INSERT INTO a SELECT generate_series(1,10);
INSERT INTO b SELECT generate_series(1,10);
DROP TABLE a, b;
RESET search_path;
DROP SCHEMA s6;
